JNI: Do not create 8M block cache for negative blockCacheSize values

javeme opened the pull request:
As [BlockBasedTableConfig setBlockCacheSize()](https://github.com/facebook/rocksdb/blob/1966a7c055f6e182d627275051f5c09441aa922d/java/src/main/java/org/rocksdb/BlockBasedTableConfig.java#L728) said, If cacheSize is non-positive, then cache will not be used. but when we configure a negative number or 0, there is an unexpected result: the block cache becomes 8M.

- Allow 0 as a valid size. When block cache size is 0, an 8MB block cache is created, as it is the default C++ API behavior. Also updated the comment.
- Set no_block_cache true if negative value is passed to block cache size, and no block cache will be created.

sagar0 commented:
0 is a valid size to block cache and I feel in such a case we should fall back to the default behavior on the C++ API side, which is to create a 8MB block cache when block_cache is set to nullptr. 
If you don't want the block cache to be used, you need to use setNoBlockCache(true). 

https://github.com/facebook/rocksdb/blob/6d113fc066c5816887eb19c84d12c0677a68af2b/include/rocksdb/table.h#L129-L136

javeme commented:
@sagar0 @adamretter Thanks for the review, I have updated the code:

1. allowed 0 as a valid size and updated the comment.
2. set no_block_cache true if negative number is passed.

adamretter commented:
@javeme LGTM
